packages/config-loader/src/schema/compile.ts

```diff
@@ -101,17 +108,38 @@ export function compileConfigSchemas(
 
   const merged = mergeConfigSchemas(schemas.map(_ => _.value));
 
-  if (options?.noUndeclaredProperties) {
-    traverse(merged, (schema: SchemaObject) => {
-      /**
-       * The `additionalProperties` key can only be applied to `type: object` in the JSON
-       *  schema.
-       */
-      if (schema?.type === 'object') {
-        schema.additionalProperties ||= false;
+  traverse(
+    merged,
+    (
+      schema: SchemaObject,
+      _: string,
+      _1?: any,
+      _2?: any,
+      _3?: any,
+      parentSchema?: SchemaObject,
+    ) => {
+      // Inherit parent deepVisibility if we don't define one ourselves.
+      if (parentSchema?.deepVisibility) {
+        schema.deepVisibility ??= parentSchema?.deepVisibility;
```

Comment:
What do you think about detecting and rejecting conflicts here? I.e. if there is a mix of frontend and secret here then we reject the schema?

Reply:
I think that makes sense. Was trying to think if there were any use cases where that might be an issue, but couldn't come up with anything.

Reply:
We can always easily disable that check too in the future, but it's way harder to add it later